test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,468 @@
+
+    /**
+     * This is used to setup the various services required by the system test framework.
+     */
+    @Environment
+    public static void initialize() throws MarathonException {
+        URI zkUri = startZookeeperInstance();
+        startBookkeeperInstances(zkUri);
+        URI controllerUri = ensureControllerRunning(zkUri);
+        ensureSegmentStoreRunning(zkUri, controllerUri);
+    }
+
+    @Before
+    public void setup() {
+        controllerURI = fetchControllerURI();
+        streamManager = StreamManager.create(Utils.buildClientConfig(controllerURI));
+
+        connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+        controller = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(Utils.buildClientConfig(controllerURI))
+                .build(), connectionFactory.getInternalExecutor());
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        controller.close();
+        connectionFactory.close();
+    }
+
+    @Test(timeout = 60000)
+    public void getNextStreamCutWithScaleUpTest() throws SegmentTruncatedException, ExecutionException, InterruptedException, TimeoutException {
+        String streamName = "testStreamSegment";
+        String streamScope = "testScopeSegment";
+        String readerGroupName = "testReaderGroupSegment";
+        String readerName = UUID.randomUUID().toString();
+
+        StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+        ClientConfig clientConfig = ClientConfig.builder()
+                .controllerURI(fetchControllerURI())
+                .build();
+        URI controllerURI = clientConfig.getControllerURI();
+
+        Stream stream = Stream.of(streamScope, streamName);
+
+        @Cleanup
+        ClientFactoryImpl clientFactory = new ClientFactoryImpl(streamScope, controller, connectionFactory);
+
+        assertTrue(controller.createScope(streamScope).join());
+        assertTrue(controller.createStream(streamScope, streamName, config).join());
+
+        @Cleanup
+        BatchClientFactory batchClient = BatchClientFactory.withScope(streamScope, clientConfig);
+
+        List<SegmentRange> ranges = Lists.newArrayList(batchClient.getSegments(stream, StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        List<Segment> list = ranges.stream().map(SegmentRange::getSegment).collect(Collectors.toList());
+        assertEquals(1, list.size());
+        log.info("Segment name ::{}", list.get(0).getScopedName());
+
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+
+        // write events to stream 30 *10  = 300 bytes
+        write30ByteEvents(10, writer);
```

Comment:
Why incorporating the size of the event in the method name? The event size should be a constant that could be changed, so we can test that this works for other sizes too.